packages/@aws-cdk/aws-servicediscovery/test/test.instance.ts

```diff
@@ -0,0 +1,355 @@
+import { expect, haveResource } from '@aws-cdk/assert';
+import ec2 = require('@aws-cdk/aws-ec2');
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+import cdk = require('@aws-cdk/cdk');
+import { Test } from 'nodeunit';
+import servicediscovery = require('../lib');
+
+export = {
+  'Instance for service in HTTP namespace'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    const namespace = new servicediscovery.Namespace(stack, 'MyNamespace', {
+      name: 'http',
+      type: servicediscovery.NamespaceType.Http,
+    });
+
+    const service = new servicediscovery.Service(stack, 'MyService', {
+      namespace,
+    });
+
+    new servicediscovery.Instance(stack, 'MyInstance', {
```

Comment:
If I understand correctly, instance attributes are very different between the namespace types, so I think there is actually value in having three different `Namespace` objects and then propagate this strong-typing down the line. 

Eventually, would it make sense to have some sugar methods like:

```ts
const namespace = new HttpNamespace(this, 'MyHttpNamespace');
const cardsService = namespace.addService('Cards');
cardsService.addInstance('41332780-d796-feed-face-02252334a661', {
  ipv4: '10.0.0.0',
  ipv6: '...',
  port: 443
});
```

Reply:
Discussing with @rix0rrr we thought having a single INamespace interface that returned the different types of L1 Namespace constructs would be sufficient. I have a commit that removes the individual namespace classes that I forgot to cherry-pick off my other branch.

Reply:
Hmm, okay. As long as @rix0rrr is cool with it, I'll put the three different Namespace L2s back after all.